src/DataProviders/SqlDataProvider/SqlDataProvider.ts

```diff
@@ -0,0 +1,68 @@
+﻿//import sql from "mssql";
+const sql = require('mssql');
+import * as assert from "assert";
+import { IDataProvider } from "../IDataProvider";
+import { ConnectionPool, ConnectionError, config, IResult } from "mssql";
+import { SeasonInfo } from "../../DataModels/SeasonInfo";
+
+/*
+ * Sql Data Provider
+ */
+export class SqlDataProvider implements IDataProvider {
+    private _tablePrefix: string = "Hymns_";
+
+    constructor(private _config: config) {
+    }
+
+    private _getQualifiedName(sp: string): string {
+        return `${this._tablePrefix}${sp}`
+    }
+
+    //TODO: why is the return type requiring void?
+    public getSeasonList(): Promise<void | SeasonInfo[]> {
```

Comment:
[async..await](https://basarat.gitbooks.io/typescript/docs/async-await.html) will make your life way easier :D

Reply:
you can replace all the `.then(..).catch(..)` with just simple `await foo();` and remove a lot of nesting in here.